### PR TITLE
Fix get cded dem res

### LINKED ISSUE #15: Should `res` in `get_cded_dem()` default to resolution of input template?

### PR DESCRIPTION
**Pull request by gcperk**
Hi @ateucher 
I reworked this function to allow output resolution to be controlled by default template or user defined output (param = res, as a numeric (in meters).

I might be missing something in the if else statements, as I converted default res to a TRUE /FALSE, for initial check but I think this may skip over the is.numeric() check. It is a bit cumbersome as is with series of if statements. 

Tests passing, but failing on internet download issue as discussed. 

This should closes #15.  

Thanks

### CODE REVIEW

File: R/get_cded_dem.R [hunk 1]
```diff
@@ -19,33 +20,70 @@
...
+#' get_cded_dem(
+#'   aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
+#'   res = FALSE,
+#'   out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_abs,
+#'   write_output = TRUE
+#'   overwrite = FALSE)
 #' }
 get_cded_dem <- function(aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
-                         res = 5,
+                         res = FALSE,
```
Comment (gcperk) [inline R/get_cded_dem.R:31]:
```suggestion
                         res = NULL,
```

File: R/get_cded_dem.R [hunk 2]
```diff
@@ -4,11 +4,12 @@
...
 #' create_raster_template().
 #'
-#' @param  aoi the `SpatRast` created using the create_base_raster() on which the
-#'      DEM data will be extracted, currently accepts raster (.tif) or filepath.
-#' @param res resolution in meters of final project
-#' @param write_output should the snapped aoi bounding box be written to disk?
-#'     If `TRUE` (default), will write to `out_dir`
+#' @param  aoi An `SpatRast` object or path to a spatial file (.tif) usually generated
+#'      when calling [create_template_raster()]. Should be a meter based coordinate reference system.
+#' @param res resolution in meters of final project. If no values specified (default is FALSE),
```
Comment (gcperk) [inline R/get_cded_dem.R:9]:
```suggestion
#' @param res resolution in meters of final project. If no values specified (default is `NULL`),
```

File: R/get_cded_dem.R [hunk 3]
```diff
@@ -19,33 +20,70 @@
...
+  cded <- terra::project(cded_raw, aoi_template)
 
-  ## check if folder exists
   if (write_output) {
+
+    output_dir <- fs::path(out_dir, paste0(output_res, "m"))
+
     if (!fs::dir_exists(fs::path(output_dir))) {
       fs::dir_create(fs::path(output_dir), recurse = TRUE)
     }
```
Comment (ateucher) [inline R/get_cded_dem.R:89]:
```suggestion
    fs::dir_create(fs::path(output_dir), recurse = TRUE)
```

`dir_create()` silently fails if dir already exists, so don't need to check its existence

File: R/get_cded_dem.R [hunk 4]
```diff
@@ -19,33 +20,70 @@
 #'
 #' @examples
 #' \dontrun{
-#' get_cded_dem(aoi_bb = file.path(fid$shape_dir_1010[2],"aoi_snapped.gpkg"),
-#'     res = 5, out_dir = fid$cov_dir_1020[2]))
+#' get_cded_dem(
+#'   aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
+#'   res = FALSE,
+#'   out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_abs,
+#'   write_output = TRUE
```
Comment (ateucher) [inline R/get_cded_dem.R:27]:
```suggestion
#'   write_output = TRUE,
```

**Review by ateucher — APPROVED** [review]
Looks good Gen! I have a suggestion to simplify some of the res logic in get_cded_dem(), but it's too complicated for an inline suggestion in the review. Do you mind if I push it?

**Comment by ateucher** [comment]
Hey @gcperk - I pushed a change where I simplified the logic of dealing with the `res` argument and creating the template aoi, and pulled it into a helper function.

I also added another test so we test both default res, and a new res. Let me know what you think

File: R/get_cded_dem.R [hunk 5]
```diff
@@ -19,33 +20,38 @@
 #'
 #' @examples
 #' \dontrun{
-#' get_cded_dem(aoi_bb = file.path(fid$shape_dir_1010[2],"aoi_snapped.gpkg"),
-#'     res = 5, out_dir = fid$cov_dir_1020[2]))
+#' get_cded_dem(
+#'   aoi = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_abs, "25m", "template.tif"),
+#'   res = FALSE,
```
Comment (gcperk) [inline R/get_cded_dem.R:25]:
```suggestion
#'   res = NULL,
```

**Comment by gcperk** [comment]
@ateucher - Thanks - that is great. I like the res check and it seems much cleaner, I might even use for other function down the line.

**Pull request merged**